### DISCUSSION TITLE
Select & Surround - works for brackets (round, square, curly) what about other characters?

### DISCUSSION THREAD

**Discussion by ghost** [zed-industries/zed#10746] [Feature Requests]
So I noticed that you can select a word, then type any of the brackets, and Zed will surround the selected word with those brackets. I looked through the setting file to see if there was a way to make Zed do this for other characters, such as: single-quote, double-quote, angle-bracket, single/double underscore, single/double asterisk, backtick. Some of these could be useful when working with Markdown.

**Comment by osiewicz** [comment]
This behaviour is controlled on language-by-language basis. Current list of characters for Markdown is available in https://github.com/zed-industries/zed/blob/main/crates/languages/src/markdown/config.toml#L6 ; there's no user controlled way to change that list (other than recompiling Zed yourself with that list changed).

**Reply by ibolit** [reply]
Got here to ask basically that same question but in a different key: I would like to be able to disable that select and surround behaviour. It would be great if this feature was user-configurable.

**Reply by BKG01** [reply]
Just new to zed , I haven't found the change surround commands yet (vim) can anyone point me in the direction if it's in the docs or post here? Thanks.

**Reply by fyodorio** [reply]
Isn't there an API that'd allow to make it through an extension? For me, as for some others, it's a second nature in writing markdown docs, tbh.

**Comment by mocha** [comment]
+1 to this -- I write a lot of markdown for docs, and in particular I use select-and-surround for `_`/`*` a lot. Would be very nice to have.

**Comment by madwort** [comment]
+1 - I use this select-surround with backtick a lot when writing docs in Markdown & comments in python (using TextMate)

**Comment by kurokirasama** [comment]
it would be nice to be able to configure this

**Reply by janxkoci** [reply]
If it worked like in Sublime text (language-agnostic), I wouldn't _need_ to configure this :sweat_smile:

**Comment by wanradt** [comment]
I think the standard pairs (<tt>'',"",(),[],{}</tt>) could be available as a base, not dependent on languages (like it works in Sublime Text and VS Code). The possibility of defining language-based additional pairs would be nice.

**Reply by janxkoci** [reply]
All these currently work for me in markdown, as well as `**` (not `__` though). They also work in a bunch of scripts I tried (several languages).

**Comment by Deepcode007** [comment]
@MartinYe1234 and Others, please add this feature to support all pairs or add a feature to make it user configurable.
I have tried this but it doesn't work.
I really need the <> brackets asap. Would be nice if add others too or make it user configurable.
`
{

    "context": "Editor",
    "bindings": {
        "shift <": [
            "editor::InsertSnippet",
            {
                "snippet": "<$ZED_SELECTED_TEXT>$0"
            }
        ]
    }
  }
`

Thanks!

**Reply by janxkoci** [reply]
Have you tried setting a syntax for the buffer (file), as we have discussed above? It mostly works for me with the files I tried..

**Reply by Deepcode007** [reply]
How do I do that? I need this feature in cpp.
